ALUM/ALUM/Services/UserService.swift

```diff
@@ -33,140 +33,62 @@ struct MentorPostData: Codable {
-            // Check the response status code
+struct MenteeGetData: Decodable {
+    var message: String
+    var mentee: MenteeInfo
+}
 
-            guard let httpResponse = response as? HTTPURLResponse else {
-                throw APIError.networkError()
-            }
+struct MentorGetData: Decodable {
+    var message: String
+    var mentor: MentorInfo
+}
 
-            if httpResponse.statusCode != 201 {
-                let responseStr = String(decoding: responseData, as: UTF8.self)
-                throw APIError.invalidRequest(
-                    message: "Error { code: \(httpResponse.statusCode), message: \(responseStr) }"
-                )
-            } else {
-                print("POST \(url) was successful.")
-            }
-        } catch {
-            print(error)
-            throw error
-        }
-    }
+/// This class deals with all communications with mentor / mentee routes on the backend.
+/// Each public method of this service
+class UserService {
+    static let shared = UserService()
 
     func createMentee(data: MenteePostData) async throws {
+        let route = APIRoute.postMentee
+        var request = try await route.createURLRequest()
         guard let jsonData = try? JSONEncoder().encode(data) else {
-            throw APIError.invalidRequest(
-                message: "Failed to Encode Data"
-            )
+            throw AppError.internalError(.jsonParsingError, message: "Failed to Encode Data")
         }
-        return try await self.createUser(url: APIRoutes.menteePOST, jsonData: jsonData)
+        request.httpBody = jsonData
+        let responseData = try await ServiceHelper.shared.sendRequestWithSafety(route: route, request: request)
+        print("SUCCESS - \(route.label)")
     }
 
     func createMentor(data: MentorPostData) async throws {
+        let route = APIRoute.postMentor
+        var request = try await route.createURLRequest()
         guard let jsonData = try? JSONEncoder().encode(data) else {
-            throw APIError.invalidRequest(
-                message: "Failed to Encode Data"
-            )
-        }
-        return try await self.createUser(url: APIRoutes.mentorPOST, jsonData: jsonData)
-    }
-    func getCurrentAuth() async throws -> String? {
-        if let currentUser = Auth.auth().currentUser {
-            do {
-                let tokenResult = try await currentUser.getIDTokenResult()
-                return tokenResult.token
-            } catch let error {
-                // Handle the error
-                throw APIError.authenticationError(
-                    message: "Error getting auth token: \(error.localizedDescription)"
-                )
-            }
-        } else {
-            // User is not logged in
-            print("User is not logged in")
-            return nil
+            throw AppError.internalError(.jsonParsingError, message: "Failed to Encode Data")
```

Comment:
Error throwing is very important because if we don't do this and in future our app is crashing we will have to debug where the error is coming from